### PR TITLE
gnome3: always use latest vala 

### PR DESCRIPTION
**Pull request by worldofpeace**
I forgot to do this when adding vala_0_46

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by worldofpeace** [comment]
@jtojnar Should we even keep this alias?

**Comment by jtojnar** [comment]
IMHO we should move this into the aliases section and point it to `pkgs.vala`. GNOME should be always using latest.

**Comment by worldofpeace** [comment]
> IMHO we should move this into the aliases section and point it to `pkgs.vala`. GNOME should be always using latest.

Added an alias.

**Review by jtojnar — APPROVED** [review]

File: pkgs/desktops/gnome-3/default.nix [hunk 1]
```diff
@@ -358,4 +357,6 @@ lib.makeScope pkgs.newScope (self: with self; {
   gamesPackages = throw "deprecated 2019-08-25: please use `services.gnome3.games.enable`";
 
   nautilus-sendto = throw "deprecated 2019-09-17: abandoned";
+
+  vala = pkgs.vala; # added 2019-10-10
```
Comment (jtojnar) [inline pkgs/desktops/gnome-3/default.nix]:
```suggestion
  inherit (pkgs) vala; # added 2019-10-10
```

**Comment by jtojnar** [comment]
And we also need to remove the use of aliases. Not sure why does not borg catch it.

**Comment by worldofpeace** [comment]
Don't know what's up with ofborg with merging. This works to staging just fine.

**Pull request merged**